File: DesktopSample/API/Section/Stylesheets/C#/Stylesheets/StyleSheetsForm.cs
using GrapeCity.ActiveReports;
using System;
using System.IO;
using System.Windows.Forms;
using System.Xml;

namespace ActiveReports.Samples.StyleSheets
{
	public partial class StyleSheetsForm : Form
	{
		private string _externalStyleSheet = string.Empty;

		public StyleSheetsForm()
		{
			InitializeComponent();
		}

		private void buttonRunReport_Click(object sender, EventArgs e)
		{
			XmlReader xmlReader = null;
			// 
			// ビューワに表示するレポートを選択します。
			if (radioButtonProductListReport.Checked)
				xmlReader = XmlReader.Create(Properties.Resources.ProductsReport);
			else if (radioButtonCategoriesReport.Checked)
				xmlReader = XmlReader.Create(Properties.Resources.CategoryReport);

			var report = new SectionReport();
			report.LoadLayout(xmlReader);
			// 
			// レポートにスタイルシートを適用します。
			string outputFolder = new FileInfo(GetType().Assembly.Location).DirectoryName + "\\";

			string styleSheet = "";
			if (radioButtonClassicStyle.Checked)
				styleSheet = outputFolder + "Classic.reportstyle";
			else if (radioButtonColoredStyle.Checked)
				styleSheet = outputFolder + "Colored.reportstyle";
			else if (_externalStyleSheet != "")
				styleSheet = _externalStyleSheet;

			if (styleSheet != "")
			{
				report.LoadStyles(styleSheet);
			}
			reportViewer.LoadDocument(report);
		}

		private void buttonChooseExtStyle_Click(object sender, EventArgs e)
		{
			// 
			// レポートに適用する外部のスタイルシートを選択します。
			FileDialog openFileDialog = new OpenFileDialog();
			openFileDialog.Filter = Properties.Resources.Filter;
			openFileDialog.InitialDirectory = new FileInfo(GetType().Assembly.Location).DirectoryName;
			openFileDialog.CheckFileExists = true;

			if (openFileDialog.ShowDialog(this) == DialogResult.OK)
			{
				FileInfo styleSheetFile = new FileInfo(openFileDialog.FileName);
				_externalStyleSheet = styleSheetFile.FullName;
				radioButtonExternalStyleSheet.Text = Properties.Resources.ExternalSstylesheet + styleSheetFile.Name;
				radioButtonExternalStyleSheet.Checked = true;
			}
		}
	}
}

File: DesktopSample/API/Section/Stylesheets/VB.NET/Stylesheets/StyleSheetsForm.vb
Imports System.IO
Imports System.Text
Imports System.Xml
Imports GrapeCity.ActiveReports

Public Class StyleSheetsForm

	Private externalStyleSheet As String = ""

	Sub New()
		Encoding.RegisterProvider(CodePagesEncodingProvider.Instance)
		InitializeComponent()
	End Sub

	Private Sub buttonChooseExtStyle_Click(ByVal sender As System.Object, ByVal e As EventArgs) Handles buttonChooseExtStyle.Click
		'レポートに適用する外部のスタイルシートを選択します。
		Dim openFileDialog As FileDialog = New OpenFileDialog()
		openFileDialog.Filter = My.Resources.Filter
		openFileDialog.InitialDirectory = New FileInfo(Me.GetType().Assembly.Location).DirectoryName
		openFileDialog.CheckFileExists = True

		If openFileDialog.ShowDialog(Me) = DialogResult.OK Then
			Dim styleSheetFile As FileInfo = New FileInfo(openFileDialog.FileName)
			externalStyleSheet = styleSheetFile.FullName
			radioButtonExternalStyleSheet.Text = My.Resources.ExternalSstylesheet + styleSheetFile.Name
			radioButtonExternalStyleSheet.Checked = True
		End If
	End Sub

	Private Sub buttonRunReport_Click(ByVal sender As System.Object, ByVal e As EventArgs) Handles buttonRunReport.Click
		Dim xmlReader As XmlReader = Nothing
		'ビューワに表示するレポートを選択します。
		If radioButtonProductListReport.Checked Then
			xmlReader = XmlReader.Create(My.Resources.ProductsReport)
		ElseIf radioButtonCategoriesReport.Checked Then
			xmlReader = XmlReader.Create(My.Resources.CategoryReport)
		End If

		Dim report As New SectionReport
		report.LoadLayout(xmlReader)
		'レポートにスタイルシートを適用します。
		Dim outputFolder As String
		outputFolder = New FileInfo(Me.GetType().Assembly.Location).DirectoryName & "\"

		Dim styleSheet As String = ""
		If radioButtonClassicStyle.Checked Then
			styleSheet = outputFolder & "Classic.reportstyle"
		ElseIf radioButtonColoredStyle.Checked Then
			styleSheet = outputFolder & "Colored.reportstyle"
		ElseIf externalStyleSheet <> "" Then
			styleSheet = externalStyleSheet
		End If

		If styleSheet <> "" Then
			report.LoadStyles(styleSheet)
		End If
		reportViewer.LoadDocument(report)

	End Sub

End Class
